src/snarks/groth16/powersoftau_utils.hpp

```diff
@@ -39,6 +39,33 @@ class srs_powersoftau
         const libff::G2<ppT> &beta_g2);
 };
 
+/// Encodings of lagrange polynomials at tau, computed from
+/// powersoftau, on some domain of degree n.
+class srs_lagrange_evaluations
+{
+public:
+    const size_t degree;
+
+    /// ${ [ L_i(x) ]_1 }_i$
+    std::vector<libff::G1<ppT>> lagrange_g1;
```

Comment:
Looks like these classes are not "templatized" since you have used a using instruction.
Could you please remove the `using`, rename to the concerned files as with the `tcc` extension and move the `using` directives to the entrypoint? (related comments: https://github.com/clearmatics/zeth/pull/41#discussion_r319096756 and https://github.com/clearmatics/zeth/pull/41#discussion_r319100937)

Reply:
I've templaized all the code that makes sense.  Some loading code relies on details of rust powersoftau data formats, so cannot work with any curve, but I've templatized where possible.